### PR TITLE
[detect-secrets](https://github.com/Yelp/detect-secrets) pre-commit hook

### PR DESCRIPTION
**Pull request by tonyzhao6**
Reviewer: @suzinyou 
Estimate: ~5 minutes

---

## Ticket

Fixes: N/A

### Goal

Ensure that we don't accidentally commit _**unintended**_ secrets.

### Changes

1. Updated `.gitignore` to include `.secrets.baseline`, which contains the baseline secrets that are already committed. In theory, these are the secrets we allow by default. It is worth checking this file and seeing if all of these secrets are intended to be public.
2. Updated `requirements.txt` for `detect-secrets`.
3. Updated pre-commit config.

### Future Tasks (optional)

1. `detect-secrets` is highly configurable. As we go along, we'll have a better sense of what we to include/exclude, etc. For now, this PR just establishes a baseline for secrets.

## How has this been tested?

1. [Example demonstration and explanation](https://www.loom.com/share/c4708aed1ad24fe19a76c9ae5ffc8495?sid=6d346b1f-6aa6-489d-a4e1-a4d84fafd547)

## To-do before merge (optional)

1. Try to commit some secrets in various ways and see if `detect-secrets` actually catches it!

## Checklist

Fill with `x` for completed. 

- [X] My code follows the style guidelines of this project
- [X] I have reviewed my own code to ensure good quality
- [X] I have tested the functionality of my code to ensure it works as intended
- [X] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [X] I have updated the requirements

### CODE REVIEW

**Comment by tonyzhao6** [comment]
Note that we want to commit the `.secrets.baseline` file so that everyone has the same baseline for secrets.

File: core_backend/requirements.txt
Comment (suzinyou) [inline core_backend/requirements.txt:1]: Should it be here or in aaq-core/requirements-dev.txt?

**Review by suzinyou — APPROVED** [review]
I have tested it!! Very cool. Tried to add ssh keyfiles :)
But I think it should go to requirements-dev.txt?

File: core_backend/requirements.txt
Comment (tonyzhao6) [reply core_backend/requirements.txt:1]: oh yea, requirements-dev! will make the change.

**Pull request merged**